main.py

```diff
@@ -160,94 +162,97 @@ def is_ignored(field, ignored_range):
-
-        # Check by `table_id`
-        if self.is_ignored(flow.table_id, self.tab_id_ignored_range):
-            return True
-        return False
-
     @listen_to("kytos/of_core.flow_stats.received")
     def on_flow_stats_check_consistency(self, event):
         """Check the consistency of a switch upon receiving flow stats."""
-        if not ENABLE_CONSISTENCY_CHECK:
+        self.check_consistency(event.content["switch"])
+
+    def check_consistency(self, switch):
+        """Check consistency of stored and installed flows given a switch."""
+        if not ENABLE_CONSISTENCY_CHECK or not switch.is_enabled():
             return
-        switch = event.content["switch"]
-        if switch.is_enabled():
+        with self._storehouse_lock:
+            log.debug(f"check_consistency on switch {switch.id} has started")
             self.check_storehouse_consistency(switch)
             if switch.dpid in self.stored_flows:
                 self.check_switch_consistency(switch)
+            log.debug(f"check_consistency on switch {switch.id} is done")
+
+    @staticmethod
+    def switch_flows_by_cookie(switch):
+        """Build switch.flows indexed by cookie."""
+        installed_flows = defaultdict(list)
+        for cookie, flows in itertools.groupby(switch.flows, lambda x: x.cookie):
+            for flow in flows:
+                installed_flows[cookie].append(flow)
+        return installed_flows
 
     def check_switch_consistency(self, switch):
-        """Check consistency of installed flows for a specific switch."""
+        """Check consistency of stored flows for a specific switch."""
         dpid = switch.dpid
-
-        # Flows stored in storehouse
-        stored_flows = self.stored_flows[dpid]["flow_list"]
-
         serializer = FlowFactory.get_class(switch)
+        installed_flows = self.switch_flows_by_cookie(switch)
 
-        for stored_flow in stored_flows:
-            stored_time = get_time(stored_flow.get("created_at", "0001-01-01T00:00:00"))
-            if (now() - stored_time).seconds <= STATS_INTERVAL:
-                continue
-            command = stored_flow["command"]
-            stored_flow_obj = serializer.from_dict(stored_flow["flow"], switch)
-
-            flow = {"flows": [stored_flow["flow"]]}
+        for cookie, stored_flows in self.stored_flows[dpid].items():
+            for stored_flow in stored_flows:
+                stored_time = get_time(
+                    stored_flow.get("created_at", "0001-01-01T00:00:00")
+                )
+                if (now() - stored_time).seconds <= STATS_INTERVAL:
+                    continue
+                stored_flow_obj = serializer.from_dict(stored_flow["flow"], switch)
+                if stored_flow_obj in installed_flows[cookie]:
+                    continue
 
-            if stored_flow_obj not in switch.flows:
-                if command == "add":
-                    log.info("A consistency problem was detected in " f"switch {dpid}.")
```

Comment:
I've augmented all of the `A consistency problem` to be a bit more informative what happened (if you search for `Consistency check:`, a lot of time I caught myself waiting for the next message to understand the problem. I added the reason just to it becomes clearer and more natural to what expect in the next message that will either delete or install the flow.

Reply:
Yes, great point! That will be very useful for troubleshooting